### PR TITLE
Throw PredicateMissingException when no predicate is pushed down

### PR DESCRIPTION
**Pull request by stagraqubole**
For https://github.com/qubole/Hive-JDBC-Storage-Handler/issues/12

### CODE REVIEW

File: src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/JdbcInputFormat.java [hunk 1]
```diff
@@ -159,6 +160,10 @@ public void configure(JobConf jobConf) {
 	public RecordReader<LongWritable, DbRecordWritable> getRecordReader(
 			InputSplit split, JobConf job, Reporter reporter)
 			throws IOException {
+		if (job.get(TableScanDesc.FILTER_TEXT_CONF_STR) == null &&
```
Comment (snarayananqubole) [inline src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/JdbcInputFormat.java]: do you need to check for empty string?

File: src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/Constants.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 	public static final String LIST_COLUMNS = "columns";
 	public static final String LIST_COLUMN_TYPES = "columns.types";
 
+	public static final String PREDICATE_COMPULSORY = "mapreduce.jdbc.predicate.compulsory";
```
Comment (snarayananqubole) [inline src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/Constants.java]: better param name? should indicate that this is related to jdbc storage handler

Comment (stagraqubole) [inline src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/Constants.java]: Followed the convention from DBConfiguration, parameters there for jdbc storage handler are named like "mapreduce.jdbc.input.field.names"

File: src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/JdbcInputFormat.java (hunk 1 above)
Comment (stagraqubole) [inline src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/JdbcInputFormat.java]: Will add that

File: src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/Constants.java [hunk 3]
```diff
@@ -22,6 +22,8 @@
     public static final String LIST_COLUMNS = "columns";
     public static final String LIST_COLUMN_TYPES = "columns.types";
     public static final String VPC_SPLIT_MAPPERS = "mapred.jdbc.hive.lazy.split";
+    public static final String PREDICATE_COMPULSORY = "jdbc.storage.handler.predicate.compulsory";
```
Comment (snarayananqubole) [inline src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/Constants.java]: compulsory -> required

File: src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/JdbcInputFormat.java [hunk 4]
```diff
@@ -65,6 +66,10 @@ public JdbcInputFormat() {
     @Override
     public RecordReader getRecordReader(InputSplit split, JobConf job,
             Reporter reporter) throws IOException {
+        if ((job.get(TableScanDesc.FILTER_TEXT_CONF_STR) == null || job.get(TableScanDesc.FILTER_TEXT_CONF_STR).length() == 0) &&
```
Comment (snarayananqubole) [inline src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/JdbcInputFormat.java]: long line - split it

File: src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/Constants.java [hunk 5]
```diff
@@ -22,6 +22,8 @@
     public static final String LIST_COLUMNS = "columns";
     public static final String LIST_COLUMN_TYPES = "columns.types";
     public static final String VPC_SPLIT_MAPPERS = "mapred.jdbc.hive.lazy.split";
```
Comment (snarayananqubole) [inline src/main/java/org/apache/hadoop/hive/jdbc/storagehandler/Constants.java]: can you rename this variable as well? maybe call it LAZY_SPLIT?

**Pull request merged**